### DISCUSSION TITLE
User friendly message about error regarding resource exhaustion in case of on-device

### DISCUSSION THREAD

**Discussion by gyapeee** [pieces-app/support#87] [Ideas]
Does Pieces able to determine that computer resource exhaustion happens/happened? If yes,x could we get an error message like that?

Reason of why:
I tried to run an On-device LLama2 7B on my private computer. I was not aware of that it requres huge amount of GPU memory.  When I asked a question from Pieces regarding a code snippet in IntelliJ plugin it failed. I got a similar to "somthing went wrong" error message and the Pieces OS has crashed. So it is possible that some resource exhaustion happened regarding insufficient GPU memory.

**Comment by rosie-at-pieces** [comment]
We have some updates regarding this coming in the next few releases! Keep an eye out :dancer:

**Comment by caleb-at-pieces** [comment]
Hey there @gyapeee , the plugins vscode, jetbrains, jupyterlab, and obsidian should now have an error warning shown if you try to use a model that we determine may overload your machine!

**Reply by gyapeee** [reply]
I've checked that scenario in vscode and intellij and Yes it is warning me that the VRAM is less than require. Thank you very much!
